### PR TITLE
working first pass for KubeVirt.io stage

### PR DESCRIPTION
**Pull request by zehicle**
Installs the KubeVirt.io via
1) installing qemu prereqs on all machines
2) runs the namespace from the template from the leader
3) waits for the virt-handler to be running

This checks the qemu readiness but does not stop if there are issues.

At this time, all the params are hardwired because that's how they present the install in the docs.  It should be possible to enhance this installation to respect labels, check configs and do other worthwhile checks.

### CODE REVIEW

File: krib/tasks/krib-kubevirt.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+Documentation: |
+  Installs KubeVirt on the leader.
+  This uses the Digital Rebar Cluster pattern so krib/cluster-profile must be set.
+
+  SUPPORTED FOR CENTOS ONLY AT THIS TIME
+
+  The install checks
+  #. to make sure ALL nodes in system are virtualization capable
+  #. to see if KubeVirt is running and may skip initialization. 
+  #. you may need to add `intel_iommu=on` to the kernel-console param
```
Comment (fabiand) [inline krib/tasks/krib-kubevirt.yaml]: Wise … :)

Aiming at SR-IOV?

File: krib/tasks/krib-kubevirt.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+      # Get access and who we are.
+      {{template "setup.tmpl" .}}
+
+      # make sure that we can run qemu
+      {{if (eq "centos" .Env.OS.Family) -}}
+      echo "CentOS ONLY at this time!"
+      exit 1
+      {{end -}}
+      for v in virt-install virt-top libguestfs-tools; do
```
Comment (fabiand) [inline krib/tasks/krib-kubevirt.yaml]: Why do you install these packages on the node?

KubeVirt does not have any (specific) user-space dependencies on the node. Instead everything is delivered in containers.

File: krib/tasks/krib-kubevirt.yaml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+          STAT=$(kubectl get pods -n kube-system -l=kubevirt.io=virt-handler --field-selector=status.phase=Running)
+          ESCAPE=0
+
+          # install KubeVirt
+          while [[ $ESCAPE -lt 30 && -z $STAT ]] ; do
+            kubectl apply -f kubevirt.yaml
+            STAT=$(kubectl get pods -n kube-system -l=kubevirt.io=virt-handler --field-selector=status.phase=Running)
+            echo "...waiting 10s for KubeVirt.io Handler to be running ($ESCAPE of 30)"
+            kubectl get pods -n kube-system -l=kubevirt.io=virt-handler
+            sleep 10
```
Comment (fabiand) [inline krib/tasks/krib-kubevirt.yaml]: I hope that this issue will help you:

https://github.com/kubevirt/kubevirt/issues/1454

File: krib/templates/kubevirt.cfg.tmpl [hunk 4]
```diff
@@ -0,0 +1,747 @@
+# KubeVirt v0.10.0 add-on
+# From https://github.com/kubevirt/kubevirt/releases
+# Monitoring
+apiVersion: v1
+kind: Service
+metadata:
+  name: kubevirt-prometheus-metrics
```
Comment (fabiand) [inline krib/templates/kubevirt.cfg.tmpl]: Awesome, directly including metrics!

File: krib/tasks/krib-kubevirt.yaml (hunk 2 above)
Comment (fabiand) [reply krib/tasks/krib-kubevirt.yaml]: Oh - It could be about `virt-host-validate` - but this is only needed for host validation, not at runtime.

Comment (galthaus) [reply krib/tasks/krib-kubevirt.yaml]: yes, I think that is why they are installed.

**Comment by fabiand** [comment]
Yeah, if this is slowing things down to much, then you can probably
simplify this a bit.

/me needs to finish the video to understand if this is the blocker for
coreos ..

**Comment by fabiand** [comment]
@karmab didn't you try kubevirt on coreos?

**Comment by karmab** [comment]
> @karmab didn't you try kubevirt on coreos?

yep, i did a couple of months ago

**Comment by galthaus** [comment]
I suspect the CoreOS failure is around trying to run the virt-host-validate command and not having packages to "fix" the problem.

File: krib/tasks/krib-kubevirt.yaml (hunk 3 above)
Comment (zehicle) [reply krib/tasks/krib-kubevirt.yaml]: This is great!  I'll be able to use it for our helm stages too.

File: krib/tasks/krib-kubevirt.yaml (hunk 2 above)
Comment (zehicle) [reply krib/tasks/krib-kubevirt.yaml]: The centos restriction is about reducing the scope with "yum install *" action than actually required.  We may be able to walk this back or move it into a "make virtualization host" stage.

Comment (zehicle) [reply krib/tasks/krib-kubevirt.yaml]: hmmm, if we don't require the packages then I'll back that out and just use virt-host-validate as a logging check that can fail.

**Comment by fabiand** [comment]
We could actually provide a container to do this ...

**Comment by galthaus** [comment]
> We could actually provide a container to do this ...
> 
> But after all it's not a mandatory check

@fabiand - but it could be really useful for automatically tagging worker nodes appropriately.

**Comment by zehicle** [comment]
Thanks @fabiand - I've updated the patch
1) does not install packages that are redundant 
2) verified to install on coreos - will require additional testing

In the future, would be good to have host-virt-verify in a container for testing

**Pull request merged**